uaclient/entitlements/repo.py

```diff
@@ -130,4 +135,8 @@ def operational_status(self):
         match = re.search(r'(?P<pin>(-)?\d+) %s' % repo_url, out)
         if match and match.group('pin') != APT_DISABLED_PIN:
             return status.ACTIVE, '%s is active' % self.title
+        if os.getuid() != 0 and entitlement_cfg.get('localEnabled', False):
```

Comment:
What's the rationale for only using the cache for non-root users? If it's correct for non-root users (which hopefully we're confident of!), then can we use it for root also, and save ourselves the `apt-cache` call?

Reply:
Ahh the reason I thought of limiting this case to just non-root users is because it is not the real configuration on the system and could become outdated in two cases:
 - someone manually removes  apt sources or apt auth config for a entitlement outside of ua client tooling
- the backend retires the current auth token for some reason and the invalid token is still being used. In that case ua status would lie saying enabled, but apt-get update would be choking with 403s.

Reply:
my thought is we at least want to insulate root user from such potentially stale inconsistency.

Reply:
If we are root in `operational_status`, should we update the cache once we have determined what the status is? That would increase our chances of having a cache that represents reality.

Reply:
+1 I like it. it would solve some corner cases.

Reply:
Are you happy including that in this PR, or shall we raise a separate (but still blocker) issue?

Reply:
actually let's hold off on this PR, I think that needs broader discussion Let's file a blocker issue and discuss there

Reply:
Did this issue get filed?